tracing-appender/src/rolling/compression.rs

```diff
@@ -0,0 +1,217 @@
+        match self {
+            GzipCompressionLevel::Literal(lit) => match lit {
+                GzipCompressionLevelLiteral::None => Compression::none(),
+                GzipCompressionLevelLiteral::Fast => Compression::fast(),
+                GzipCompressionLevelLiteral::Best => Compression::best(),
+            },
+            GzipCompressionLevel::Numerical(num) => Compression::new(num as u32),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct GzipCompression {
+    pub(crate) level: GzipCompressionLevel,
+}
+
+/// Data structure to pass compression parameters
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum CompressionConfig {
+    Gzip(GzipCompression),
+}
+
+impl CompressionConfig {
+    pub(crate) fn gz_compress_level(&self) -> Compression {
+        match self {
+            CompressionConfig::Gzip(gz) => {
+                let level = gz.level.clone().into();
+                level
+            }
+        }
+    }
+
+    #[allow(unused)]
+    pub(crate) fn extension(&self) -> Option<&str> {
+        match self {
+            CompressionConfig::Gzip(_) => Some("gz"),
+        }
+    }
+}
+
+/// Defines a compression level for gzip algorithm.
+///
+/// Compression levels are defined as they are in `flate2` crate where
+/// - compression level 0 (`CompressionOption::GzipNone` or `CompressionOption::GzipLevel0`)
+/// - compression level 1 (`CompressionOption::GzipFast` or `CompressionOption::GzipLevel1`)
+/// - compression level n (where n is between 2 and 9)
+/// - compression level 9 (`CompressionOption::GzipBest` or `CompressionOption::GzipLevel9`)
+///
+/// ```rust
+/// # #[cfg(feature = "gzip_compression")] {
+/// # fn docs() {
+/// use tracing_appender::compression::CompressionOption;
+/// let compression_level = CompressionOption::GzipBest;
+/// # }
+/// # }
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum CompressionOption {
+    /// No compression (gzip compression level 0)
+    GzipNone,
+    /// Fast compression (gzip compression level 1)
+    GzipFast,
+    /// Fast compression (gzip compression level 9)
+    GzipBest,
+    /// Gzip compression level 0
+    GzipLevel0,
+    /// Gzip compression level 1
+    GzipLevel1,
+    /// Gzip compression level 2
+    GzipLevel2,
+    /// Gzip compression level 3
+    GzipLevel3,
+    /// Gzip compression level 4
+    GzipLevel4,
+    /// Gzip compression level 5
+    GzipLevel5,
+    /// Gzip compression level 6
+    GzipLevel6,
+    /// Gzip compression level 7
+    GzipLevel7,
+    /// Gzip compression level 8
+    GzipLevel8,
+    /// Gzip compression level 9
+    GzipLevel9,
+}
+
+impl Into<CompressionConfig> for CompressionOption {
+    fn into(self) -> CompressionConfig {
+        match self {
+            CompressionOption::GzipNone => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Literal(GzipCompressionLevelLiteral::None),
+            }),
```

Comment:
Just a suggestion, you could define a constructor on `CompressionConfig` that accepts `GzipCompressionLevelLiteral` and then these lines would be a lot terser:

```rust
CompressionOption::GzipNone => CompressionConfig::new_gzip(GzipCompressionLevelLiteral::None),
```

or, you could define a helper closure just to use in this method, like:

``` rust
let new_gzip = |level| -> {
    CompressionConfig::Gzip(GzipCompression {
        level: GzipCompressionLevel::Literal(level),
    }
}

match self {
    CompressionOption::GzipNone => gzip_new(GzipCompressionLevelLiteral::None),
   ... // etc
```

Reply:
Fixed will push soon.